core/src/main/java/io/jenkins/pluginhealth/scoring/probes/AbstractDeprecatedAnnotationImportsCheckProbe.java

```diff
@@ -0,0 +1,83 @@
+package io.jenkins.pluginhealth.scoring.probes;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+import io.jenkins.pluginhealth.scoring.model.ProbeResult;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.CollectionUtils;
+
+/**
+ * The abstract probe checks for deprecated imports in a plugin.
+ */
+public abstract class AbstractDeprecatedAnnotationImportsCheckProbe extends Probe {
```

Comment:
I was doubtful whether I should go for an abstract class. 

Though I do not know all the problems that come with plugin maintenance, it seems that in the future a situation may arise where we might need to search in `.java` files.

Reply:
I hope this isn't an incorrect decision. The goal is to make it future proof but if you think this situation may not arise, I am open to refactoring it.

Reply:
Without entering the debate about whether we should keep it or not, I just wanted to share some interesting readings about some principles we (should?) tend to follow while writing code: https://blog.devgenius.io/key-principles-in-software-dry-kiss-yagni-solid-and-other-acronyms-98e5575a6942?gi=09e171025fdc

The key take aways in the current situation are *KISS* and *YAGNI*.

*KISS* (Keep It Simple, Stupid) is all about trying to write code in the simplest way possible.

*YAGNI* (You Ain't Gonne Need It) is slightly related and is all about wondering if you're really ever going to need what you implemented.

Again, I am not saying we should or should not go for that design you implemented, but rather sharing some general tips about coding: in general, we try to stick to the minimum change we can do in order to implement the feature that we need. Preparing stuff for a potential future is interesting only if we actually did study a bit what is about to come and we know for sure that we will need that preparation.

OTOH though, I really do appreciate both the effort, and the comments explaining your rationale. 

The decision itself is neither correct or incorrect, this is a Pull Request, so you are proposing something, with a rationale. Then we discuss it all together as a team working on the project, and we all agree on what seems to be the best decision based on everyone's arguments.

Great move on triggering that discussion Jagruti!

Reply:
Thanks for your encouraging words. 

I read the article and according to the _YAGNI_ the abstract shouldn't be a part as of now. 

>  Then we discuss it all together as a team working on the project, and we all agree on what seems to be the best decision based on everyone's arguments.

I agree with this. Whatever the decision is, it will be fine by me.